backend/workflow_manager/endpoint/destination.py

```diff
@@ -81,14 +86,41 @@ def _get_endpoint_for_workflow(
             )
         return endpoint
 
+    def _get_source_endpoint_for_workflow(
```

Comment:
NIT:  Better to move to source.py  like
@classmethod
def _get_endpoint_for_workflow(

Not mandatory

Reply:
will do later in version 2